sdk/eventhub/Azure.Messaging.EventHubs.Experimental/src/Producer/PartitionPublishingOptions.cs

```diff
@@ -0,0 +1,132 @@
+﻿// Copyright (c) Microsoft Corporation. All rights reserved.
+// Licensed under the MIT License.
+
+using System.ComponentModel;
+
+namespace Azure.Messaging.EventHubs.Experimental
+{
+    /// <summary>
+    ///   The set of options that can be specified for an <see cref="IdempotentProducer" />
+    ///   to influence its behavior when publishing directly to an Event Hub partition.
+    /// </summary>
+    ///
+    /// <remarks>
+    ///   These options are ignored when publishing to the Event Hubs gateway for automatic
+    ///   routing or when using a partition key.
+    /// </remarks>
+    ///
+    public class PartitionPublishingOptions
```

Comment:
The `PartitionPublishingXXX` classes are `internal` in the core package;  I couldn't find a reasonable to change that visibility to expose them without creating a local alias - so I did.